src/properties.cpp

```diff
@@ -215,46 +215,47 @@ std::mutex dump_lock;
  * Note that this is not checking for epsilon-validity.
  */
 bool Manifold::Impl::IsSelfIntersecting() const {
-  const double epsilonSq = epsilon_ * epsilon_;
+  const double ep = 2 * epsilon_;
```

Comment:
This is the main change - in line with our degenerate checks and such, we increase epsilon to account for error within these checks themselves. The rest of this function is just renames because I was confused by x and y not being directions.